src/controllers/productController.js

```diff
@@ -141,7 +111,7 @@ class productController{
         });
       }
   
-      if (req.user && req.user.role.roleName === 'vendor' && req.user.role.id !== product.vendorId) {
+      if (req.user && req.user.role.roleName === 'vendor' && req.user.role.id !== product.VendorId) {
```

Comment:
You see this condition is more complex so it's a good practice to add a comment to explain the logic behind the condition.